C/Cairomm/Cairomm@1.14.4/build_tarballs.jl

```diff
@@ -0,0 +1,45 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Cairomm"
+version = v"1.14.4"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://www.cairographics.org/releases/cairomm-$(version).tar.xz",
+                  "4749d25a2b2ef67cc0c014caaf5c87fa46792fc4b3ede186fb0fc932d2055158")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/cairomm*/
+mkdir output && cd output/
+meson --cross-file=${MESON_TARGET_TOOLCHAIN%.*}_gcc.meson ..
+ninja -j${nproc}
+ninja install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+
+# These platforms are not supported by Cairo_jll yet
+platforms = filter!(p -> arch(p) != "armv6l", supported_platforms())
+platforms = filter!(p -> !(Sys.isapple(p) && arch(p) == "aarch64"), platforms)
+platforms = filter!(p -> libc(p) != "musl", platforms)
+
+platforms = expand_cxxstring_abis(platforms; skip=Returns(false))
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct(["libcairomm-1", "libcairomm-1.0"], :libcairomm)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="Cairo_jll", uuid="83423d85-b0ee-5818-9007-b63ccbeb887a"), v"1.14.12"; compat="^1.12.0")
```

Comment:
#3024
```suggestion
    Dependency(PackageSpec(name="Cairo_jll", uuid="83423d85-b0ee-5818-9007-b63ccbeb887a"); compat="1.16.1")
```

Reply:
Side note, I think it's usually fishy to use both a build version and a compat spec:
```julia
Dependency("Packagename_jll", build_version; compat="<compat_spec>")
```
If you care about compatibility you should usually use for building the oldest version you claim compaitiblity with, and conversely if you use a specific version for building in general you can't expect to be actually compatible with an older version, especially on macOS where the compatibility version can create problems.  All in all I prefer to use the compat spec only, the oldest version corresponding to that range will be then automatically used for building.

Reply:
That makes sense.

I've been specifying compatibility this way to have things in terms of the full upstream version span (in particular, BinaryBuilder will error out if the bottom of your version span hasn't actually been released and registered), but I'm happy to treat this as a compat on the released BinaryBuilder artifacts and pull in the lower bound